### ISSUE TITLE
[BUG] Deleted button persists on page in some conditions

### ISSUE THREAD

**Issue by phillipivan** [bitfocus/companion#2897]
### Is this a bug in companion itself or a module?

- [X] I believe this to be a bug in companion

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

A button with an very large font size persists on the page layout after being deleted.

### Steps To Reproduce

1. Create Button
2. Set Button Text 'HI', Set Font Size to 166
3. Delete Button
4. Button remains on page but the Edit button tab shows an empty button
5. Creating a new button in the same position with a new text string is not displayed.

Screen shot showing deleted button still displayed in page view:

![image](https://github.com/bitfocus/companion/assets/137139426/eefd4d57-0e24-4f0e-a4be-22e0a2c59e53)

### Expected Behavior

Deleted button is removed from page view, and when a new button is created it is displayed in that position as expected.

### Environment (please complete the following information)

```markdown
- OS: Win 10.0.19045
- Browser: Chrome 120
- Companion Version: 3.3.1
```

### Additional context

Replicated on multiple systems.

Companion restart clears the zombie button.

**Comment by Julusian** [comment]
oh wow, this isn't a good bug.
After testing this, my companion vm has 'crashed' stuck with 100% cpu and memory usage
![image](https://github.com/bitfocus/companion/assets/1327476/cbd08689-d47e-4028-87bd-e83c59a1e5a9) ![image](https://github.com/bitfocus/companion/assets/1327476/16021820-d2dd-436a-ad2e-6f4208dc4991)

**Comment by phillipivan** [comment]
We didn't observe outright crashing.

Obviously there is no sane reason for a font size like that - it came about from a typo when setting up a button.

**Comment by Julusian** [comment]
I have a fix for this, it was hitting an edge case in the code when trying to figure out where to force line breaks in the text.
It was getting stuck in an infinite loop, finding each iteration that no characters fit on a line, so it created a line of 0 characters then kept on trying the remainder. 
So the fix is simple, if one char is too wide to draw, do it anyway instead of trying to see if zero chars is a better fit.

Unfortunately this is too late to get into 3.3.1, but the fix is in the betas and will be in a 3.3.2 when that happens

@phillipivan if you haven't restarted companion since you saw this, I highly recommend you do. One of the rendering threads will be stuck trying to complete that draw, consuming a core of cpu and as much memory as it can find.

**Comment by phillipivan** [comment]
Thanks for the quick response. Confirmed fixed for us in build 7040.
